Jacoco extensions results in java.lang.NoSuchMethodError: 'boolean[] ***.$jacocoInit()' for kotlin when panache repo is injected

## Describe the bug

In [this reproduction project](https://github.com/languitar/quarkus-jacoco-kotlin-issue) `mvn verify` results in the following test error:

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.751 s <<< FAILURE! - in org.acme.rest.GreetingResourceTest
[ERROR] org.acme.rest.GreetingResourceTest.testHelloEndpoint  Time elapsed: 0.004 s  <<< ERROR!
org.junit.jupiter.api.extension.TestInstantiationException: Failed to create test instance
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
Caused by: java.lang.reflect.InvocationTargetException
Caused by: java.lang.NoSuchMethodError: 'boolean[] org.acme.rest.FooEntityRepo.$jacocoInit()'
```

This happens in case:
* the quarkus jacoco extension is included
* AND a Panache repository is injected into a test class

Things work if only one of the above conditions is met.

### Expected behavior

Tests can be executed and coverage reports are functional.

### Actual behavior

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.751 s <<< FAILURE! - in org.acme.rest.GreetingResourceTest
[ERROR] org.acme.rest.GreetingResourceTest.testHelloEndpoint  Time elapsed: 0.004 s  <<< ERROR!
org.junit.jupiter.api.extension.TestInstantiationException: Failed to create test instance
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
Caused by: java.lang.reflect.InvocationTargetException
Caused by: java.lang.NoSuchMethodError: 'boolean[] org.acme.rest.FooEntityRepo.$jacocoInit()'
```

## To Reproduce

Steps to reproduce the behavior:
1. Clone the [reproduction project](https://github.com/languitar/quarkus-jacoco-kotlin-issue)
2. Execute  `mvn verify`

### Configuration

As included in the project

## Environment (please complete the following information):

### Output of `uname -a` or `ver`

```
Linux pres 5.10.47-1-lts #1 SMP Wed, 30 Jun 2021 13:52:19 +0000 x86_64 GNU/Linux
```

### Output of `java -version`

```
openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment (build 11.0.11+9)
OpenJDK 64-Bit Server VM (build 11.0.11+9, mixed mode)
```

### GraalVM version (if different from Java)

not used here

### Quarkus version or git rev

As in the project. 2.0.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

```
Apache Maven 3.8.1 (NON-CANONICAL_2021-04-26T21:52:54Z_root)
Maven home: /opt/maven
Java version: 11.0.11, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-11-openjdk
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.10.47-1-lts", arch: "amd64", family: "unix"
```

--- a/extensions/panache/panache-common/deployment/src/main/java/io/quarkus/panache/common/deployment/visitors/KotlinPanacheClassOperationGenerationVisitor.java
+++ b/extensions/panache/panache-common/deployment/src/main/java/io/quarkus/panache/common/deployment/visitors/KotlinPanacheClassOperationGenerationVisitor.java
@@ -537,6 +537,9 @@ public class KotlinPanacheClassOperationGenerationVisitor extends ClassVisitor {
                 .orElse(null);
         if (methodInfo != null && !methodInfo.hasAnnotation(DOTNAME_GENERATE_BRIDGE)) {
             return super.visitMethod(access, name, descriptor, signature, exceptions);
+        } else if (name.contains("$")) {
+            //some agents such as jacoco add new methods, they generally have $ in the name
+            return super.visitMethod(access, name, descriptor, signature, exceptions);
         }
         return null;
     }

